### ISSUE TITLE
[clang-tidy] bugprone-throw-keyword-missing on default member initializer: "did you mean 'throw shared_ptr'?"

### ISSUE THREAD

**Issue by N-Dekker** [llvm/llvm-project#129675]
Using LLVM 19.1.7, I encountered a false positive `bugprone-throw-keyword-missing` from [ITK](https://itk.org)'s [itkExceptionObject.h](https://github.com/InsightSoftwareConsortium/ITK/blob/32a2a6de17ffb7c8319ab38dbe61bd3b7c171f00/Modules/Core/Common/include/itkExceptionObject.h), which can be reproduced as follows:

```cpp
#include <exception>
#include <memory>

class MyException : public std::exception
{
public:
  MyException() = default;
private:
  class NestedData;
  std::shared_ptr<NestedData> m_shared_data{};
};

class Bug : public MyException
{
public:
  Bug()
  {
    // Non-defaulted default constructor.
  }
};

```

Output:

```
warning: suspicious exception object created but not thrown; did you mean 'throw shared_ptr'? [bugprone-throw-keyword-missing]
   10 |   std::shared_ptr<NestedData> m_shared_data{};
      |                                            ^
```

**Comment by N-Dekker** [comment]
Similar to commit 738e7f1231949ec248c1d8d154783338215613d1 by @FabianWolff (author) and @AaronBallman (committer), Nov 16, 2021:

```
Fix false positive in bugprone-throw-keyword-missing check

Fixes PR#52400. The tests for bugprone-throw-keyword-missing actually
already contain exceptions as class members, but not as members with
initializers, which was probably just an oversight.

```

However, that commit from 2021 must have been included with "my" LLVM 19.1.7 already.

**Comment by carlosgalvezp** [comment]
The code posted in the OP does not trigger on clang-tidy trunk:
https://godbolt.org/z/rohhnEM7v

Is it ok to close this issue?

**Comment by N-Dekker** [comment]
Thanks for double-checking at godbolt, @carlosgalvezp I think that your commit 69fb9bcde0312e672e6f2280f8662784731d79e6 has indeed already fixed the issue, and the fix should be included with LLVM 20.1. So hereby I'll close my issue 😃

**Comment by carlosgalvezp** [comment]
Great to hear, thanks! 😄
